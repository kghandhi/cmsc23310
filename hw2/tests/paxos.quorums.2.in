# Example with two proposers and five acceptors.
#
# Same as paxos.quorums.1 but with Proposer 1 recovering.
# However, it gets stuck because it has only received
# one ACCEPTED message (which is actually a delayed message
# send before Proposer 1 failed) and two REJECTED messages,
# so it cannot conclude whether there is consensus (or
# whether the proposed value was rejected)

2 5 40
0 FAIL ACCEPTOR 4
0 FAIL ACCEPTOR 5
0 PROPOSE 1 42
10 FAIL PROPOSER 1
10 FAIL ACCEPTOR 1
10 FAIL ACCEPTOR 2
10 RECOVER ACCEPTOR 4
10 RECOVER ACCEPTOR 5
13 PROPOSE 2 0
28 RECOVER PROPOSER 1
0 END
