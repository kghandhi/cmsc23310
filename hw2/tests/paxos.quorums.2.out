  0: ** A4 FAILS **
  0: ** A5 FAILS **
  0:    -> P1  PROPOSE v=42
  1: P1 -> A1  PREPARE n=1
  2: P1 -> A2  PREPARE n=1
  3: P1 -> A3  PREPARE n=1
  4: A1 -> P1  PROMISE n=1 (Prior: None)
  5: A2 -> P1  PROMISE n=1 (Prior: None)
  6: A3 -> P1  PROMISE n=1 (Prior: None)
  7: P1 -> A1  ACCEPT n=1 v=42
  8: P1 -> A2  ACCEPT n=1 v=42
  9: P1 -> A3  ACCEPT n=1 v=42
 10: ** P1 FAILS **
 10: ** A1 FAILS **
 10: ** A2 FAILS **
 10: ** A4 RECOVERS **
 10: ** A5 RECOVERS **
 11:
 12:
 13:    -> P2  PROPOSE v=0
 14: P2 -> A3  PREPARE n=2
 15: P2 -> A4  PREPARE n=2
 16: P2 -> A5  PREPARE n=2
 17: A3 -> P2  PROMISE n=2 (Prior: n=1, v=42)
 18: A4 -> P2  PROMISE n=2 (Prior: None)
 19: A5 -> P2  PROMISE n=2 (Prior: None)
 20: P2 -> A3  ACCEPT n=2 v=42
 21: P2 -> A4  ACCEPT n=2 v=42
 22: P2 -> A5  ACCEPT n=2 v=42
 23: A3 -> P2  ACCEPTED n=2 v=42
 24: A4 -> P2  ACCEPTED n=2 v=42
 25: A5 -> P2  ACCEPTED n=2 v=42
 26:
 27:
 28: ** P1 RECOVERS **
 28: P1 -> A4  PREPARE n=1
 29: P1 -> A5  PREPARE n=1
 30: P1 -> A4  ACCEPT n=1 v=42
 31: P1 -> A5  ACCEPT n=1 v=42
 32: A3 -> P1  ACCEPTED n=1 v=42
 33: A4 -> P1  REJECTED n=1
 34: A5 -> P1  REJECTED n=1
 35:
 36:
 37:
 38:
 39:
 40:

P1 did not reach consensus
P2 has reached consensus (proposed 0, accepted 42)
